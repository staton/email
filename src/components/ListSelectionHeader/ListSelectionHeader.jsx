import React, {Component} from 'react';
import PropTypes from 'prop-types';
import HeaderButton from '../HeaderButton/HeaderButton';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import { 
    MdArrowBack, 
    MdDelete  } 
from 'react-icons/md';
import { 
    selectEmail,
    setInboxListItemsActive 
} from '../../redux/actions/email-actions';

const propTypes = {
    leftElementGroupClassName: PropTypes.string.isRequired,
    rightElementGroupClassName: PropTypes.string.isRequired,
    elementClassName: PropTypes.string.isRequired
};

const defaultProps = {
};

export class ListSelectionHeader extends Component {

    constructor() {
        super();

        this.handleBackButtonClicked = this.handleBackButtonClicked.bind(this);
        this.handleDeleteButtonClicked = this.handleDeleteButtonClicked.bind(this);
    }

	render() {
        const backButtonId = 'ListSelectionHeader__back-button';
        const deleteButtonId = 'ListSelectionHeader__delete-button';

		return (
			<header className="ListSelectionHeader">
                <div className={this.props.leftElementGroupClassName}>
                    <div className={this.props.elementClassName}>
                        <HeaderButton
                            content={<MdArrowBack />}
                            id={backButtonId}
                            onClick={this.handleBackButtonClicked} 
                        />
                    </div>
                </div>
                <div className={this.props.rightElementGroupClassName}>
                    <div className={this.props.elementClassName}>
                        <HeaderButton
                            content={<MdDelete />}
                            enabled={this.props.selectedEmails.length > 0}
                            id={deleteButtonId}
                            onClick={this.handleDeleteButtonClicked} 
                        />
                    </div>
                </div>
			</header>
		);
    }

    /**
     * Called when the user clicks the back button.
     * @param {object} e The event.
     */
    handleBackButtonClicked(e) {
        this.props.selectEmail([], false);
        this.props.setInboxListItemsActive(false);
    }

    /**
     * Called when the user clicks the delete button.
     * @param {object} e The event.
     */
    handleDeleteButtonClicked(e) {
        console.log('delete clicked');
    }

}

ListSelectionHeader.propTypes = propTypes;
ListSelectionHeader.defaultProps = defaultProps;

function mapStateToProps(store, ownProps) {
    return {
        selectedEmails: store.email.selectedEmails
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
        selectEmail: selectEmail,
        setInboxListItemsActive: setInboxListItemsActive
    },
    dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(ListSelectionHeader);