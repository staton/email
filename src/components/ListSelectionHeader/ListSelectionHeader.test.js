import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { ListSelectionHeader } from './ListSelectionHeader';
import Email from '../../models/email';

describe('<ListSelectionHeader />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(
            <ListSelectionHeader
                leftElementGroupClassName="Header__left-element-group"
                rightElementGroupClassName="Header__right-element-group"
                elementClassName="Header__element"
                selectedEmails={[]}
            />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('has a back button and a delete button', () => {
        const wrapper = shallow(
            <ListSelectionHeader
                leftElementGroupClassName="Header__left-element-group"
                rightElementGroupClassName="Header__right-element-group"
                elementClassName="Header__element"
                selectedEmails={[]}
            />);
        expect(wrapper.find('#ListSelectionHeader__back-button').length).toEqual(1);
        expect(wrapper.find('#ListSelectionHeader__delete-button').length).toEqual(1);
    });
    
});