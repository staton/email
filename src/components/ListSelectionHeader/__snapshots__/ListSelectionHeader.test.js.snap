// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListSelectionHeader /> renders correctly 1`] = `
<header
  className="ListSelectionHeader"
>
  <div
    className="Header__left-element-group"
  >
    <div
      className="Header__element"
    >
      <HeaderButton
        content={<Unknown />}
        enabled={true}
        id="ListSelectionHeader__back-button"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="Header__right-element-group"
  >
    <div
      className="Header__element"
    >
      <HeaderButton
        content={<Unknown />}
        enabled={false}
        id="ListSelectionHeader__delete-button"
        onClick={[Function]}
      />
    </div>
  </div>
</header>
`;
